src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -135,13 +143,50 @@ const useCoreChatSystemHook = () => {
     return websocketData;
   };
 
+  const helperWebsocketForDeletedMessage = async (websocketData: GetstreamWebsocket) => {
+    const websocketMessage = websocketData?.message;
+    const isDeletedMessage = websocketMessage?.message_type === MESSAGE_TYPE_DELETED;
+
+    if (isDeletedMessage) {
+      const selectedChat = await ChatSchema.getByid(
+        localDb,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        websocketMessage.deleted_message_id!
+      );
+      const selectedCreatedAt = selectedChat.createdAt;
+      const selectedUpdatedAt = selectedChat.updatedAt;
+
+      websocketMessage.id = websocketMessage.deleted_message_id ?? '';
+      websocketMessage.text = DELETED_MESSAGE_TEXT;
+      websocketMessage.created_at = selectedCreatedAt ?? websocketMessage.created_at;
+      websocketMessage.updated_at = selectedUpdatedAt ?? websocketMessage.updated_at;
+
+      ChatSchema.updateDeletedRepliedChat(
+        localDb,
+        websocketData?.channel_id,
+        websocketMessage.deleted_message_id ?? '',
+        selectedChat.createdAt
+      );
+
+      const {replyTarget} = replyPreview;
+      if (replyTarget && replyTarget?.id === websocketMessage?.deleted_message_id) {
+        const newReplyPreview = {...replyTarget};
+        newReplyPreview.message = DELETED_MESSAGE_TEXT;
+        newReplyPreview.message_type = MESSAGE_TYPE_DELETED;
+        setReplyTarget(newReplyPreview, dispatch);
+      }
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function `helperWebsocketForDeletedMessage` seems to handle multiple responsibilities such as updating the chat schema, setting reply targets, and modifying websocket messages. This could violate the Single Responsibility Principle (SRP). Consider breaking this function into smaller ones, each handling a single responsibility.



<!-- This is an auto-generated comment by OSS CodeRabbit -->